src/main/java/racing/RacingCarList.java

```diff
@@ -0,0 +1,64 @@
+package racing;
+
+import racing.strategy.MoveStrategy;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class RacingCarList {
+
+    private List<RacingCar> racingCarList;
+
+    public RacingCarList(String[] input, MoveStrategy strategy){
+            this.racingCarList = RacingCarList.setRacingCars(input, strategy);
+    }
+
+    public static List<RacingCar> setRacingCars(String[] input, MoveStrategy strategy) {
+        List<RacingCar> result = new ArrayList<RacingCar>();
+        Arrays.stream(input).forEach( name -> result.add(new RacingCar(strategy, name)) );
+
+        return result;
+    }
```

Comment:
이 부분도 마찬가지로 MoveStrategy를 생성자에서 주입할 필요는 없을 것 같아요!
그리고 다음과 같이 표현할 수 있습니다.

```suggestion
    public RacingCarList(List<RacingCar> racingCars){
            this.racingCarList = racingCars;
    }

    public static RacingCarList of(List<RacingCar> racingCars) {
        return new RacingCarList(racingCars);
    }

    public static RacingCarList of(String[] input, MoveStrategy strategy) {
        return Arrays.stream(input)
                     .map(new RacingCar(strategy, name))
                     .collect(collectingAndThen(toList(), RacingCarList::of))
    }
```

moveStrategy를 생성자에서 주입하는 게 아니라 Move Method에서 주입하다고 가정하면 이렇게 표현할 수 있습니다.

```java
    public static RacingCarList of(String[] input) {
        return Arrays.stream(input)
                     .map(RacingCar::new)
                     .collect(collectingAndThen(toList(), RacingCarList::of))
    }
```